File: SRPTests/TestScripts/WriteToUav.py
# Test for writing to a UAV from a compute shader.
# This is a special compute-only test script. It relies on stuff
# from the test framework so will not work inside Syrup itself.
from SRPScripting import *

cs = ri.CompileShader("ComputeTest.hlsl", "WriteToUAV", "cs_4_0")

numElements = 16
buffer = ri.CreateBuffer(numElements * 4, Format.R32_Float, None, uav = True)
cs.FindUavVariable("OutUAV").Set(buffer)

expected = [2.0 * i + 10.0 for i in xrange(0, numElements)]

SetExpected(expected)
SetResultBuffer(buffer)

def RenderFrame(context):
	context.Dispatch(cs, 1, 1, 1)

ri.SetFrameCallback(RenderFrame)

File: SRPTests/TestScripts/CS/WriteToUav.cs
// Test for writing to a UAV from a compute shader.
// This is a special compute-only test script. It relies on stuff
// from the test framework so will not work inside Syrup itself.
using SRPScripting;
using System.Linq;

var cs = ri.CompileShader("ComputeTest.hlsl", "WriteToUAV", "cs_4_0");

int numElements = 16;
var buffer = ri.CreateUninitialisedBuffer(numElements * 4, 4);
cs.FindUavVariable("OutUAV").Set(buffer.CreateUav());

var expected = Enumerable.Range(0, numElements)
	.Select(i => 2.0f * i + 10.0f)
	.Cast<object>();

dynamic SetExpected = _testParams["SetExpected"];
dynamic SetResultBuffer = _testParams["SetResultBuffer"];

SetExpected(expected);
SetResultBuffer(buffer);

ri.SetFrameCallback(context =>
{
	context.Dispatch(cs, 1, 1, 1);
});
